packages/components/src/custom-select-control-v2/types.ts

```diff
@@ -42,13 +35,134 @@ export type CustomSelectProps = {
 	 *
 	 * @default 'default'
 	 */
-	size?: 'default' | 'small';
+	size?: 'compact' | 'default' | 'small';
 	/**
 	 * Can be used to externally control the value of the control.
 	 */
 	value?: string | string[];
 };
 
+export type _CustomSelectProps = {
+	/**
+	 * The child elements. This should be composed of CustomSelectItem components.
+	 */
+	children: React.ReactNode;
+	/**
+	 * Used to visually hide the label. It will always be visible to screen readers.
+	 *
+	 * @default false
+	 */
+	hideLabelFromVision?: boolean;
+	/**
+	 * Label for the control.
+	 */
+	label: string;
+};
+
+export type CustomSelectProps = _CustomSelectProps &
+	Exclude< CustomSelectButtonProps, { size: 'small' } > & {
```

Comment:
I'm still seeing `'small'` as an option in Storybook, and I'm not getting an error when adding it to the new CustomSelect. 🤔 Is `Exclude` not the correct method?

Reply:
Try [this one](https://github.com/WordPress/gutenberg/pull/57902#discussion_r1471802171)!

Reply:
Ah, sorry I missed that; thank you!